converter/dicom-cast/src/Microsoft.Health.DicomCast.Core.UnitTests/Features/Worker/FhirTransaction/Patient/PatientPipelineStepTests.cs

```diff
@@ -191,6 +201,57 @@ public void GivenARequestToUpdateAPatient_WhenResponseIsOK_ThenItShouldBeNoOp()
         _patientPipeline.ProcessResponse(context);
     }
 
+    [Fact]
+    public async Task GivenIssuerIdFlagEnabled_WhenResponseisOK_ThenPatientSystemIdShouldBeEmpty()
+    {
+        FhirTransactionContext context = CreateFhirTransactionContext();
+
+        _patientConfiguration = new PatientConfiguration() { PatientSystemId = "patientSystemId", IsIssuerIdUsed = true };
+        IOptions<PatientConfiguration> optionsConfiguration = Options.Create(_patientConfiguration);
+
+        _patientPipeline = new PatientPipelineStep(_fhirService, _patientSynchronizer, optionsConfiguration, NullLogger<PatientPipelineStep>.Instance);
+
+        await _patientPipeline.PrepareRequestAsync(context, DefaultCancellationToken);
+
+        FhirTransactionRequestEntry actualPatientEntry = context.Request.Patient;
+
+        Assert.Equal("identifier=|p1", actualPatientEntry.Request.IfNoneExist);
+    }
+
+    [Fact]
+    public async Task GivenIssuerIdFlagNotPresent_WhenResponseisOK_ThenPatientSystemIdShouldBePopulatedWithConfiguredValue()
+    {
+        FhirTransactionContext context = CreateFhirTransactionContext();
+
+        _patientConfiguration = new PatientConfiguration() { PatientSystemId = "patientSystemId" };
+        IOptions<PatientConfiguration> optionsConfiguration = Options.Create(_patientConfiguration);
+
+        _patientPipeline = new PatientPipelineStep(_fhirService, _patientSynchronizer, optionsConfiguration, NullLogger<PatientPipelineStep>.Instance);
+
+        await _patientPipeline.PrepareRequestAsync(context, DefaultCancellationToken);
+
+        FhirTransactionRequestEntry actualPatientEntry = context.Request.Patient;
+
+        Assert.Equal("identifier=patientSystemId|p1", actualPatientEntry.Request.IfNoneExist);
+    }
+
+    [Fact]
+    public async Task GivenIssuerIdFlagDisabledAndPatientSystemIdNotConfigured_WhenResponseisOK_ThenPatientSystemIdShouldBeEmpty()
+    {
+        FhirTransactionContext context = CreateFhirTransactionContext();
+
+        _patientConfiguration = new PatientConfiguration() { IsIssuerIdUsed = false };
```

Comment:
Is there a test where `IsIssuerIdUsed` is `true` and we pluck the value from the tag successfully?

Reply:
It was present with an empty issuer id. But I have updated it to a more meaningful value in test `GivenIssuerIdFlagEnabledAndTagPresent_WhenResponseisOK_ThenPatientSystemIdShouldBeIssuerId`

Reply:
I don't see this test. Was it added?

Reply:
I missed adding that one. Please check now.